web/packages/teleport/src/Discover/useDiscover.tsx

```diff
@@ -522,12 +523,27 @@ export type KubeMeta = BaseMeta & {
   kube: Kube;
 };
 
-// KubeMeta describes the fields for a kube resource
-// that needs to be preserved throughout the flow.
+/**
+ * EksMeta describes the fields for a kube resource
+ * that needs to be preserved throughout the flow.
+ */
 export type EksMeta = BaseMeta & {
   kube: Kube;
 };
 
+/**
+ * AppMeta describes the fields for a app resource
+ * that needs to be preserved throughout the flow.
+ */
+export type AppMeta = BaseMeta & {
+  app: App;
+  /**
+   * list of arns extracted from an app from
+   * its field 'awsRoles: AwsRole[]'
+   */
+  awsRoleArns: string[];
```

Comment:
So we have the same exact data in `app.awsRoles`? If so, we shouldn't add another property that just copies them, because:

* you don't have a single source of truth, 
* when setting up a test/mock you have to fill both `awsRoles` and `awsRoleArns`.